File: src/main/java/com/zfy/component/basic/foundation/interceptors/AbstractInterceptor.java
package com.zfy.component.basic.foundation.interceptors;

import android.support.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * CreateAt : 2018/1/12
 * Describe : base Interceptor
 *
 * @author chendong
 */
public abstract class AbstractInterceptor implements Interceptor {

    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        return proceedResponse(chain.proceed(proceedRequest(chain.request())));
    }

    protected @NonNull
    Request proceedRequest(Request request) {
        return request;
    }

    protected @NonNull
    Response proceedResponse(Response response) {
        return response;
    }
}

File: src/main/java/com/zfy/component/basic/foundation/api/interceptors/AbstractInterceptor.kt
package com.zfy.component.basic.foundation.api.interceptors

import io.reactivex.annotations.NonNull
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

/**
 * CreateAt : 2018/9/8
 * Describe :
 *
 * @author chendong
 */
open class AbstractInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        return try {
            proceedResponse(chain.proceed(proceedRequest(chain.request())))
        } catch (e: Exception) {
            chain.proceed(chain.request())
        }
    }

    @NonNull
    open fun proceedRequest(request: Request): Request {
        return request
    }

    @NonNull
    open fun proceedResponse(response: Response): Response {
        return response
    }
}